api/v1alpha4/condition_consts.go

```diff
@@ -201,3 +201,14 @@ const (
 	// from making any further remediations.
 	TooManyUnhealthyReason = "TooManyUnhealthy"
 )
+
+// Conditions and condition Reasons for  MachineDeployments
+
+const (
+	// AvailableCondition means the MachineDeployment is available, ie. at least the minimum available
+	// machines required are up and running for at least minReadySeconds.
+	AvailableCondition ConditionType = "Available"
```

Comment:
If we want this to be machineDeployment specific should probably rename the variable to `MachineDeploymentAvailableCondition` consistently with `ControlPlaneReadyCondition` or `BootstrapReadyCondition`. Otherwise it should probably live in this block https://github.com/kubernetes-sigs/cluster-api/pull/4625/files#diff-9bdb2e3a1ad45026b8684d8d8dab63c66ac79269293891a64b6684451356764dL22

Reply:
I'm happy to change that but I'm following the convention used for KCP where the condition is simply called`AvailableCondition`. Similarly, the `ReadyCondition` is used for several objects.

Reply:
Given this condition has a specific semantic, let use a different constant (Ready condition has the same semantic across the entire board)